hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/store/contracts/TokenAccessorImpl.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hedera.node.app.service.mono.store.contracts;
+
+import static com.hedera.node.app.service.mono.store.models.NftId.fromGrpc;
+import static com.hedera.node.app.service.mono.utils.EntityIdUtils.accountIdFromEvmAddress;
+import static com.hedera.node.app.service.mono.utils.EntityIdUtils.tokenIdFromEvmAddress;
+import static com.hedera.node.app.service.mono.utils.EvmTokenUtil.convertToEvmKey;
+import static com.hedera.node.app.service.mono.utils.MiscUtils.asKeyUnchecked;
+
+import com.google.protobuf.ByteString;
+import com.hedera.node.app.service.evm.store.contracts.precompile.codec.CustomFee;
+import com.hedera.node.app.service.evm.store.contracts.precompile.codec.EvmKey;
+import com.hedera.node.app.service.evm.store.contracts.precompile.codec.EvmNftInfo;
+import com.hedera.node.app.service.evm.store.contracts.precompile.codec.EvmTokenInfo;
+import com.hedera.node.app.service.evm.store.contracts.precompile.codec.TokenKeyType;
+import com.hedera.node.app.service.evm.store.tokens.TokenAccessor;
+import com.hedera.node.app.service.evm.store.tokens.TokenType;
+import com.hedera.node.app.service.mono.ledger.properties.TokenProperty;
+import com.hedera.node.app.service.mono.store.models.NftId;
+import com.hederahashgraph.api.proto.java.NftID;
+import java.util.List;
+import java.util.Optional;
+import org.hyperledger.besu.datatypes.Address;
+
+public class TokenAccessorImpl implements TokenAccessor {
```

Comment:
We can make all arguments `final` for better readability and eliminate the chance to modify the address before fetching its metadata.